### PR TITLE
[SPARK-7980] [SQL] Support SQLContext.range(end)

### PR DESCRIPTION
**Pull request by animeshbaranawal**
Used previous defined range(start,end) to define range(end)

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -701,6 +701,17 @@ class SQLContext(@transient val sparkContext: SparkContext)
   /**
    * :: Experimental ::
    * Creates a [[DataFrame]] with a single [[LongType]] column named `id`, containing elements
+   * in an range from 0 to `end`(exclusive) with step value 1.
+   *
+   * @since 1.4.0
+   * @group dataframe
+   */
+  @Experimental
+  def range(end: Long): DataFrame = range(0,end)
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: can you add a space after comma?

**Comment by rxin** [comment]
@animeshbaranawal this looks good for Scala/Java, except one minor style issue.

Can you also update to include Python? You can find the file in python/pyspark/sql/context.py.  Thanks.

**Comment by rxin** [comment]
Jenkins, test this please.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala (hunk 1 above)
Comment (animeshbaranawal) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: ok will do it

**Comment by animeshbaranawal** [comment]
@rxin 
To include the range(end) functioning in python sql context, the only way I could figure out is to re-order the function arguments in range which is not desirable. 
Would it be correct to create another function with a different name for the purpose?

**Comment by rxin** [comment]
It is ok. You can just use start itself, i.e. if there is only one argument, then start means end.

**Pull request closed without merging**